x-pack/plugins/alerting/server/saved_objects/migrations.ts

```diff
@@ -869,3 +900,40 @@ function pipeMigrations(...migrations: AlertMigration[]): AlertMigration {
   return (doc: SavedObjectUnsanitizedDoc<RawRule>) =>
     migrations.reduce((migratedDoc, nextMigration) => nextMigration(migratedDoc), doc);
 }
+
+/**
+ * This creates a migration map that applies search source migrations to legacy es query rules
+ */
+function getEsQueryAlertSearchSourceMigrations(
+  searchSourceMigrations: MigrateFunctionsObject
+): MigrateFunctionsObject {
+  return mapValues<MigrateFunctionsObject, MigrateFunction>(
+    searchSourceMigrations,
+    (
+        migrateSerializedSearchSourceFields: MigrateFunction<SerializedSearchSourceFields>
+      ): MigrateFunction =>
+      (state) => {
+        const _state = state as { attributes: RawRule };
+
+        const serializedSearchSource = _state.attributes.params
+          .searchConfiguration as SerializedSearchSourceFields;
+
+        const isNotSerializedSearchSource =
+          typeof serializedSearchSource !== 'object' ||
+          serializedSearchSource === null ||
+          Array.isArray(serializedSearchSource);
+        if (isNotSerializedSearchSource) return _state;
```

Comment:
Sorry, I clearly didn't communicate this well. 😆 

I meant something like this to avoid the casting:
```

/**
 * This creates a migration map that applies search source migrations to legacy es query rules
 */
function getEsQueryAlertSearchSourceMigrations(
  searchSourceMigrations: MigrateFunctionsObject
): MigrateFunctionsObject {
  return mapValues<MigrateFunctionsObject, MigrateFunction>(
    searchSourceMigrations,
    (
        migrateSerializedSearchSourceFields: MigrateFunction<SerializedSearchSourceFields>
      ): MigrateFunction =>
      (state) => {
        const _state = state as { attributes: RawRule };

        const serializedSearchSource = _state.attributes.params.searchConfiguration;

        if (isNotSerializedSearchSource(serializedSearchSource)) {
          return {
            ..._state,
            attributes: {
              ..._state.attributes,
              params: {
                ..._state.attributes.params,
                searchConfiguration: migrateSerializedSearchSourceFields(serializedSearchSource),
              },
            },
          };
        }
        return _state;
      }
  );
}

function isNotSerializedSearchSource(
  maybeSerializedSearchSource: unknown
): maybeSerializedSearchSource is maybeSerializedSearchSourceFields {
  return (
    typeof maybeSerializedSearchSource !== 'object' ||
    maybeSerializedSearchSource === null ||
    Array.isArray(maybeSerializedSearchSource)
  );
}
```


This isn't necessarily _safer_, but it's more maintainable.
Also woth noting I think `isNotSerializedSearchSource` probably belongs in `src/plugins/data/common/search/search_source/types.ts` and not in `x-pack/plugins/alerting/server/saved_objects/migrations.ts`.

Reply:
Suggestion were applied.

Reply:
🙏